components/match2/wikis/heroes/match_group_input_custom.lua

```diff
@@ -0,0 +1,577 @@
+
+		-- Retrieve icon for team
+		if opponent.type == Opponent.team then
+			opponent.icon, opponent.icondark = opponentFunctions.getIcon(opponent.template)
+		end
+
+		-- apply status
+		opponent.score = string.upper(opponent.score or '')
+		if Logic.isNumeric(opponent.score) then
+			opponent.score = tonumber(opponent.score)
+			opponent.status = STATUS_SCORE
+			isScoreSet = true
+		elseif Table.includes(ALLOWED_STATUSES, opponent.score) then
+			opponent.status = opponent.score
+			opponent.score = NOT_PLAYED_SCORE
+		end
+
+		-- get players from vars for teams
+		if opponent.type == Opponent.team then
+			if not Logic.isEmpty(opponent.name) then
+				match = MatchGroupInput.readPlayersOfTeam(match, opponentIndex, opponent.name)
+			end
+		elseif Opponent.typeIsParty(opponent) then
+			opponent.match2players = Json.parseIfString(opponent.match2players) or {}
+			opponent.match2players[1] = opponent.match2players[1] or {}
+			opponent.match2players[1].name = opponent.match2players[1].name or opponent.name
+		elseif opponent.type ~= Opponent.literal then
+			error('Unsupported Opponent Type "' .. (opponent.type or '') .. '"')
+		end
+
+		opponents[opponentIndex] = opponent
+	end
+
+	matchFunctions.applyWalkover(match, opponents)
+	matchFunctions.checkIfFinished(match, isScoreSet, opponents)
+
+
+	-- apply placements and winner if finshed
+	if
+		not Logic.isEmpty(match.winner) or
+		Logic.readBool(match.finished) or
+		CustomMatchGroupInput.placementCheckSpecialStatus(opponents)
+	then
+		match.finished = true
+		match, opponents = CustomMatchGroupInput.getResultTypeAndWinner(match, opponents)
+	end
+
+	-- Update all opponents with new values
+	for opponentIndex, opponent in pairs(opponents) do
+		match['opponent' .. opponentIndex] = opponent
+	end
+	return match
+end
+
+function matchFunctions.applyWalkover(match, opponents)
+	--apply walkover input
+	match.walkover = string.upper(match.walkover or '')
+	if Logic.isNumeric(match.walkover) then
+		local winnerIndex = tonumber(match.walkover)
+		opponents = matchFunctions._makeAllOpponentsLoseByWalkover(opponents, STATUS_DEFAULT_LOSS)
+		opponents[winnerIndex].status = STATUS_DEFAULT_WIN
+		match.finished = true
+	elseif Logic.isNumeric(match.winner) and Table.includes(ALLOWED_STATUSES, match.walkover) then
+		local winnerIndex = tonumber(match.winner)
+		opponents = matchFunctions._makeAllOpponentsLoseByWalkover(opponents, match.walkover)
+		opponents[winnerIndex].status = STATUS_DEFAULT_WIN
+		match.finished = true
+	end
+end
+
+function matchFunctions.checkIfFinished(match, isScoreSet, opponents)
+	-- see if match should actually be finished if bestof limit was reached
+	match.finished = Logic.readBool(match.finished)
+		or isScoreSet and (
+			Array.any(opponents, function(opponent) return tonumber(opponent.score or 0) > match.bestof/2 end)
+			or Array.all(opponents, function(opponent) return tonumber(opponent.score or 0) == match.bestof/2 end)
```

Comment:
```suggestion
			Array.any(opponents, function(opponent) return (tonumber(opponent.score) or 0) > match.bestof/2 end)
			or Array.all(opponents, function(opponent) return (tonumber(opponent.score) or 0) == match.bestof/2 end)
```
Right? Otherwise it could error on non-numeric input